main.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/moby/moby/pkg/term"
+)
+
+const (
+	shimName    = "kata-shim"
+	exitFailure = 1
+)
+
+var shimLog = logrus.WithFields(logrus.Fields{
+	"name": shimName,
+	"pid":  os.Getpid(),
+})
+
+func initLogger(logLevel string) error {
+	shimLog.Logger.Formatter = &logrus.TextFormatter{TimestampFormat: time.RFC3339Nano}
+
+	level, err := logrus.ParseLevel(logLevel)
+	if err != nil {
+		return err
+	}
+
+	logrus.SetLevel(level)
+
+	return nil
+}
+
+func main() {
+	var (
+		logLevel      string
+		agentAddr     string
+		container     string
+		pid           uint
+		proxyExitCode bool
+		init          bool
+	)
+
+	flag.StringVar(&logLevel, "log", "warn", "set shim log level: debug, info, warn, error, fatal or panic")
+	flag.StringVar(&agentAddr, "agent", "", "agent gRPC socket endpoint")
+
+	flag.StringVar(&container, "container", "", "container id for the shim")
+	flag.UintVar(&pid, "pid", 0, "process id for the shim")
+	flag.BoolVar(&init, "init", true, "first container process in a sandbox")
+	flag.BoolVar(&proxyExitCode, "proxy-exit-code", true, "proxy exit code of the process")
+
+	flag.Parse()
+
+	if agentAddr == "" || container == "" || pid == 0 {
+		shimLog.Error("container ID, process ID and agent socket endpoint must be set")
+		os.Exit(exitFailure)
+	}
+
+	err := initLogger(logLevel)
+	if err != nil {
+		shimLog.Errorf("invalid log level(%s):%s", logLevel, err)
+		os.Exit(exitFailure)
+	}
+
+	shim, err := newShim(agentAddr, container, uint32(pid))
+	if err != nil {
+		shimLog.Errorf("failed to create new shim: %v", err)
+		os.Exit(exitFailure)
+	}
+
+	// wait to be waken up if starting a new sandbox
```

Comment:
Sorry to insist on that one, but could you please describe why we need to wait from the shim when being a sandbox shim. And also document who's supposed to send us `SIGUSR1`, and at which point.

Reply:
I dropped it because with current `StartContainer()` and `ExecProcess()` APIs, runtime can no longer start shim and create sandbox in parallel.